### PR TITLE
Use https instead of http

### PR DESCRIPTION
**Pull request by komainu8**
Dear, developers.
I've fixed an issue #1459.

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ sections:
       - title: "Language Spec"
         description: "Scala's formal language specification."
         icon: "fa fa-book"
-        link: http://scala-lang.org/files/archive/spec/2.12/
+        link: https://scala-lang.org/files/archive/spec/2.12/
```
Comment (Philippus) [inline index.md]: Could you change the link to the spec for scala 2.13: `https://scala-lang.org/files/archive/spec/2.13/` ?

**Comment by komainu8** [comment]
@Philippus Thank you for your suggestion.
I've fixed it.

File: index.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ sections:
       - title: "Language Spec"
         description: "Scala's formal language specification."
         icon: "fa fa-book"
-        link: http://scala-lang.org/files/archive/spec/2.12/
+        link: https://scala-lang.org/files/archive/spec/2.13/
```
Comment (exoego) [inline index.md]: Could you apply same changes to `_ja/index.md` too?
That file is  Japanese translation of `index.md`

**Comment by komainu8** [comment]
@exoego Yes. I applied the same change to the Japanese page too.

**Comment by exoego** [comment]
@komainu8 Thanks !!

**Comment by Philippus** [comment]
Your commit message uses version 2.1.3, this should be 2.13. You may want to change it to have a nicer history.

**Comment by komainu8** [comment]
@Philippus Thank you for your comment.
I've fixed my commit message(version 2.1.3 -> version 2.13).

**Comment by exoego** [comment]
It seems that there is only one 404 in https://ci.scala-lang.org/scala/docs.scala-lang/374/3

> - ./_site/ja/overviews/parallel-collections/ctries.html
>  *  External link https://lampwww.epfl.ch/~prokopec/ctries-snapshot.pdf failed: 404 No error

**Comment by komainu8** [comment]
@exoego Thank you for your comment.
I found also the other same broken links. So, I've also fixed them.

**Comment by komainu8** [comment]
I've finished modifying http to https(except not exist https site) in all document.

**Comment by SethTisue** [comment]
this is much more broad than just fixing #1459; it changes all links, instead of only links within scala-lang.org

that makes me a bit nervous. should I be nervous? if the link-checker likes it, then maybe it's okay...?

**Comment by komainu8** [comment]
Ah...I see.
You're right. This PR is broad.
I reduce modify.

**Comment by SethTisue** [comment]
> I reduce modify.

well, maybe it's fine as-is, actually.

any other maintainers in this repo have an opinion?

**Comment by SethTisue** [comment]
thank you @komainu8 !!

**Pull request merged**